mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ParameterDslDeclarationEnricher.java

```diff
@@ -140,7 +175,16 @@ public void visitUnion(UnionType unionType) {
     }
 
     boolean allowsInlineAsWrappedType(MetadataType type, TypeCatalog typeCatalog) {
-      return extensionDeclaration.getSubTypes().stream().anyMatch(s -> s.getBaseType().equals(type))
+      final Optional<String> typeIdOptional = getTypeId(type);
+
+      return extensionDeclaration.getSubTypes()
```

Comment:
extract this to a separate method with a descriptive name. This code in unreadable